fetch-in-react-practice/client/src/components/Todos.js

```diff
@@ -0,0 +1,153 @@
+/* eslint-disable no-unused-vars -- Remove me */
+import React, { useEffect, useState } from 'react';
+import PageTitle from './PageTitle';
+import TodoList from './TodoList';
+import TodoForm from './TodoForm';
+
+export default function Todos() {
+  const [todos, setTodos] = useState([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [error, setError] = useState();
+
+  /* Implement useEffect to fetch all todos. Hints are at the bottom of the file. */
+  useEffect(() => {
+    async function getTodos() {
+      try {
+        const res = await fetch('/api/todos');
+        if (!res.ok) {
+          throw new Error(`fetch Error ${res.status}`);
+        }
+        const data = await res.json();
+        setTodos(data);
+      } catch (error) {
+        setError(error);
+      } finally {
+        setIsLoading(false);
+      }
+    }
+    getTodos();
+  }, []);
+
+  /* Implement addTodo to add a new todo. Hints are at the bottom of the file. */
+  async function addTodo(newTodo) {
+    try {
+      const newRes = await fetch('/api/todos', {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify(newTodo),
+      });
+      if (!newRes.ok) {
+        throw new Error(`fetch Error ${newRes.status}`);
+      }
+      const newData = await newRes.json();
+      setTodos([...todos, newData]);
+    } catch (error) {
+      setError(error);
+    } finally {
+      setIsLoading(false);
+    }
+  }
+
+  /* Implement toggleCompleted to toggle the completed state of a todo. Hints are at the bottom of the file. */
+  async function toggleCompleted(todoId) {
+    let obj;
+    for (let i = 0; i < todos.length; i++) {
+      if (todos[i].todoId === todoId) {
+        obj = { isCompleted: !todos[i].isCompleted };
+      }
+    }
+    try {
+      const res = await fetch(`/api/todos/${todoId}`, {
+        method: 'PATCH',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify(obj),
+      });
+      if (!res.ok) {
+        throw new Error(`fetch Error ${res.status}`);
+      }
+      const data = await res.json();
+      const newNew = todos.map((el) => {
+        if (el.todoId === todoId) {
+          return data;
+        }
+        return el;
+      });
```

Comment:
This can be written more succinctly using `?:` and the shortcut arrow function syntax.